### PR TITLE
[SEDONA-115] Add ST_XMax and ST_XMin functions

### PR DESCRIPTION
**Pull request by kvpankaj1999**
## Is this PR related to a proposed Issue?
Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/projects/SEDONA/issues/SEDONA-115.

## What changes were proposed in this PR?
Added ST_XMax and ST_XMin functions

## How was this patch tested?
Yes, I have added the unit tests.

## Did this PR include necessary documentation updates?
Yes, I have added the documentation necessary.

### CODE REVIEW

File: sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala [hunk 1]
```diff
@@ -335,6 +335,18 @@ class functionTestScala extends TestBaseScala with Matchers with GeometrySample
 
     }
 
+    it("Passed ST_XMax") {
+      var test = sparkSession.sql("SELECT ST_XMax(ST_GeomFromWKT('POLYGON ((-3 -3, 3 -3, 3 3, -3 3, -3 -3))'))")
```
Comment (jiayuasu) [inline sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala:339]: Please use a more significant test case here. E.g., POLYGON ((-1 -11, 0 10, 1 11, 2 12, -1 -11))

File: sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala [hunk 2]
```diff
@@ -335,6 +335,18 @@ class functionTestScala extends TestBaseScala with Matchers with GeometrySample
...
+      assert(test.take(1)(0).get(0).asInstanceOf[Double] == 3.0)
+
+    }
+
+    it("Passed ST_XMin") {
+      var test = sparkSession.sql("SELECT ST_XMin(ST_GeomFromWKT('POLYGON ((-3 -3, 3 -3, 3 3, -3 3, -3 -3))'))")
```
Comment (jiayuasu) [inline sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala:345]: Please use a more significant test case here. E.g., POLYGON ((-1 -11, 0 10, 1 11, 2 12, -1 -11))

File: flink/src/test/java/org/apache/sedona/flink/FunctionTest.java [hunk 3]
```diff
@@ -88,4 +88,22 @@ public void testAsEWKT() {
         String result = (String) first(polygonTable).getField(0);
         assertEquals("POLYGON ((-0.5 -0.5, -0.5 0.5, 0.5 0.5, 0.5 -0.5, -0.5 -0.5))", result);
     }
+
+    @Test
+    public void testXMax() {
+        Table polygonTable = createPolygonTable(1);
+        Table MaxTable = polygonTable.select(call(Functions.ST_XMax.class.getSimpleName(), $(polygonColNames[0])));
+        double result = (double) first(MaxTable).getField(0);
+        System.out.println(result);
```
Comment (jiayuasu) [inline flink/src/test/java/org/apache/sedona/flink/FunctionTest.java:97]: No print to console is allowed. Please remove this line.

File: flink/src/test/java/org/apache/sedona/flink/FunctionTest.java [hunk 4]
```diff
@@ -88,4 +88,22 @@ public void testAsEWKT() {
...
+        assertEquals(0.5, result,0);
+    }
+
+    @Test
+    public void testXMin() {
+        Table polygonTable = createPolygonTable(1);
+        Table MinTable = polygonTable.select(call(Functions.ST_XMin.class.getSimpleName(), $(polygonColNames[0])));
+        double result = (double) first(MinTable).getField(0);
+        System.out.println(result);
```
Comment (jiayuasu) [inline flink/src/test/java/org/apache/sedona/flink/FunctionTest.java:106]: No print to console is allowed. Please remove this line.

File: docs/api/sql/Function.md [hunk 5]
```diff
@@ -1105,4 +1105,42 @@ Result:
...
+Since: `v1.2.1`
+
+Example:
+
+```SQL
+SELECT ST_XMax(df.geometry) AS xmax
+FROM df
+```
+
+Input: `POLYGON ((-1 1, 1 -1, -1 -1, 1 1))`
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:1125]: Polygon example here is not a WKT format. Use POLYGON ((-1 -11, 0 10, 1 11, 2 12, -1 -11)) instead.

Please fix other places in the docs

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java [hunk 6]
```diff
@@ -110,4 +111,36 @@ public String eval(@DataTypeHint(value = "RAW", bridgedTo = org.locationtech.jts
...
+        }
+
+    }
+
+    public static class ST_XMin extends ScalarFunction {
+        @DataTypeHint("Double")
+        public Double eval(@DataTypeHint(value = "RAW", bridgedTo = org.locationtech.jts.geom.Geometry.class) Object o) {
+            Geometry geom = (Geometry) o;
+            Coordinate[] coord = geom.getCoordinates();
+            double min = Double.MIN_VALUE;
```
Comment (jiayuasu) [inline flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java:136]: This one is a bug. Fix it with MaxValue

File: sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala (hunk 1 above)
Comment (kvpankaj1999) [reply sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala]: Changed the test case

File: sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala (hunk 2 above)
Comment (kvpankaj1999) [reply sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala]: Changed the test case

File: sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala (hunk 1 above)
Comment (kvpankaj1999) [reply sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala]: Changed the test case.

File: sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala (hunk 2 above)
Comment (kvpankaj1999) [reply sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala]: Changed the test case.

File: flink/src/test/java/org/apache/sedona/flink/FunctionTest.java (hunk 3 above)
Comment (kvpankaj1999) [reply flink/src/test/java/org/apache/sedona/flink/FunctionTest.java]: Removed the statement

File: flink/src/test/java/org/apache/sedona/flink/FunctionTest.java (hunk 4 above)
Comment (kvpankaj1999) [reply flink/src/test/java/org/apache/sedona/flink/FunctionTest.java]: Removed the statement

File: docs/api/sql/Function.md (hunk 5 above)
Comment (kvpankaj1999) [reply docs/api/sql/Function.md]: Changed example in all docs

File: flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java (hunk 6 above)
Comment (kvpankaj1999) [reply flink/src/main/java/org/apache/sedona/flink/expressions/Functions.java]: Changed to MaxValue.

**Comment by kvpankaj1999** [comment]
Updated the Code review changes.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Comment by jiayuasu** [comment]
Please add JIRA url to your description just like what other PRs do

**Comment by kvpankaj1999** [comment]
> Please add JIRA url to your description just like what other PRs do

I have added the JIRA URL in the description.

**Pull request closed without merging**